features/machine/machine.feature

```diff
@@ -429,3 +429,23 @@ Feature: Machine features testing
     And the output should contain:
       | Forbidden: publicIP is not allowed in Azure disconnected installation |
 
+
+  # @author miyadav@redhat.com
+  # @case_id OCP-33047
+  @admin
+  @destructive
+  Scenario: Machine API Operator should go degraded if any pod controller is crash looping
+    Given I have an IPI deployment
+    When I switch to cluster admin pseudo user
+    Then I use the "openshift-machine-api" project
+
+    And I run the steps 20 times or exit when co machine-api is degraded:
+    """
+    Given a pod becomes ready with labels:
+      | api=clusterapi, k8s-app=controller |
+    When admin ensure "<%= pod.name %>" pod is deleted from the "openshift-machine-api" project
+    And the step should succeed
+    """
+    When 180 seconds have passed
+    And evaluation of `cluster_operator('machine-api').condition(type: 'Degraded')` is stored in the :co_degraded clipboard
```

Comment:
It seems to me that you don't need a new step. You can just put the above step in the loop. So you delete pod, then check if API is degraded. 

Then wait 180 seconds and check it is still degrated.

Reply:
I mean you put these steps inside `I wait(?: up to #{NUMBER} seconds)? for the steps to pass:`